### ISSUE TITLE
[TorchInductor] `cat_splitwithsizes_replace` misses equivalent negative dimension spelling

### ISSUE THREAD

**Issue by nikokurumi** [pytorch/pytorch#196905]
### 🐛 Describe the bug

`cat_splitwithsizes_replace` removes a `cat` immediately followed by a `split_with_sizes` when the split recovers the original cat inputs. However, the pattern rejects an equivalent negative split dimension.

## Reproducer

```python
from __future__ import annotations

import torch


class Model(torch.nn.Module):
    def forward(self, a: torch.Tensor, b: torch.Tensor):
        joined = torch.cat([a, b], dim=1)
        return torch.split(joined, [3, 5], dim=-1)


def main() -> None:
    if not torch.cuda.is_available():
        raise RuntimeError("CUDA is required for this test")

    torch.manual_seed(1875)
    device = torch.device("cuda")
    a = torch.randn(2, 3, device=device)
    b = torch.randn(2, 5, device=device)
    model = Model().to(device).eval()

    with torch.no_grad():
        expected = model(a, b)
        actual = torch.compile(model, backend="inductor", fullgraph=True)(a, b)

    torch.cuda.synchronize()
    torch.testing.assert_close(actual, expected)


if __name__ == "__main__":
    main()
```

For these rank-2 inputs, `dim=-1` and `dim=1` select the same logical axis, so the split exactly recovers the original `cat` inputs.

## Observed behavior

Before post-grad:

```text
cat = aten.cat.default([a, b], 1)
split_with_sizes = aten.split_with_sizes.default(cat, [3, 5], -1)
getitem = split_with_sizes[0]
getitem_1 = split_with_sizes[1]
return (getitem, getitem_1)
```

After post-grad, the graph is unchanged:

```text
cat = aten.cat.default([a, b], 1)
split_with_sizes = aten.split_with_sizes.default(cat, [3, 5], -1)
getitem = split_with_sizes[0]
getitem_1 = split_with_sizes[1]
return (getitem, getitem_1)
```

Since `dim=-1` refers to the same axis as the `cat` dimension here, this `cat` followed by `split_with_sizes` can be replaced with the original inputs, but the optimization is missed.

## Root cause

The check compares the raw integer dimension arguments:

https://github.com/pytorch/pytorch/blob/f07882e293662058329c946feb5615b1aaeedb82/torch/_inductor/fx_passes/post_grad.py#L1777-L1780

For this case, it compares `-1` with `1` and rejects the match even though they refer to the same logical axis. The dimensions need to be canonicalized before comparison.

The reverse `splitwithsizes_cat_replace` pattern appears to have the same raw-dimension comparison.

### Versions

PyTorch main commit `f07882e`

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by Rudra-G-23** [comment]
I’d be happy to work on this once it’s considered actionable.

My plan would be to canonicalize the `cat` and `split_with_sizes` dimensions before comparing them, likely using the existing `canonicalize_dim` helper with the tensor rank. I’d also update the reverse `split_with_sizes -> cat` pattern and add regression tests covering equivalent positive/negative dimensions such as `1` and `-1`.

Please let me know if this approach looks correct.

**Comment by he-yufeng** [comment]
Fixed in #196943. Both `is_valid_cat_splitwithsizes` and its sibling `is_valid_splitwithsizes_cat` compared the cat and split dims with a plain `!=`, so `cat(dim=1)` + `split_with_sizes(dim=-1)` on a 2D input was rejected. The fix normalizes both dims against the input rank before comparing; true mismatches still reject. One subtlety caught by the new tests: the split side's rank must be read from the split's *input* node, since the split node's own meta value is a list of per-part tensors, not a `Tensor`.
